### PR TITLE
Fix array conversion from column value

### PR DESCRIPTION
**Pull request by Setogit**
Requires compatible fix of datasource juggler test case.

### CODE REVIEW

File: lib/cassandra.js [hunk 1]
```diff
@@ -625,6 +625,18 @@ Cassandra.prototype.fromColumnValue = function(prop, val) {
         break;
       case 'List':
       case 'Array':
+        if (typeof val === 'string') val = JSON.parse(val);
+        var ret = [];
+        val.forEach(function(v) {
```
Comment (raymondfeng) [inline lib/cassandra.js]: You probably need to test if `val` is an array.

File: lib/cassandra.js [hunk 2]
```diff
@@ -625,6 +625,18 @@ Cassandra.prototype.fromColumnValue = function(prop, val) {
         break;
       case 'List':
       case 'Array':
+        if (typeof val === 'string') val = JSON.parse(val);
```
Comment (raymondfeng) [inline lib/cassandra.js]: `JSON.parse` might throw an error.

File: lib/cassandra.js (hunk 1 above)
Comment (Setogit) [inline lib/cassandra.js]: done

File: lib/cassandra.js (hunk 2 above)
Comment (Setogit) [inline lib/cassandra.js]: done

File: lib/cassandra.js [hunk 3]
```diff
@@ -625,10 +611,29 @@ Cassandra.prototype.fromColumnValue = function(prop, val) {
...
+        if (typeof val === 'string') {
+          try {
+            val = JSON.parse(val);
+          } catch(e) {}
+        }
+        if (Array.isArray(val)) {
+          var ret = [];
+          val.forEach(function(v) {
+            try {
+              ret = ret.concat(JSON.parse(v));
```
Comment (raymondfeng) [inline lib/cassandra.js]: If the val is already an array, why do we try to parse items inside the array again? I don't think we need to handle such mixed usage.

Comment (Setogit) [inline lib/cassandra.js]: That turned out to be a cover-up of a bug in toColumnValue. Fixed.

**Pull request merged**